app/mixins/waffle-route.ts

```diff
@@ -0,0 +1,19 @@
+import Mixin from '@ember/object/mixin';
+import { inject as service } from '@ember/service';
+
+export default Mixin.create({
+    features: service(),
+    currentUser: service(),
+    beforeModel(...args) {
+        return this.get('currentUser._setWaflle').perform().then(() => this._super(...args));
+    },
+    actions: {
+        didTransition(...args) {
```

Comment:
Will try it! `willTransititon` was the ideal place for it, but it would've required it to go on the origin route, rather than the destination that cared about the flag, which seemed wrong. Did not think about the router's willTransition though.